man/man/man3/ConfigTheme.3
.TH "ConfigTheme" 3 "Thu Aug 23 2012" "Version 1.0" "FS-UAE Gui for Linux OS" \" -*- nroff -*-
.ad l
.nh
.SH NAME
ConfigTheme \- 
.PP
This class is responsible for the Theme Configuration management\&.  

.SH SYNOPSIS
.br
.PP
.PP
\fC#include <configtheme\&.h>\fP
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBConfigTheme\fP ()"
.br
.ti -1c
.RI "string \fBgetThemeConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetThemeString\fP ()"
.br
.ti -1c
.RI "string \fBgetFadeColorConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetFadeColorString\fP ()"
.br
.ti -1c
.RI "string \fBgetWallColor1ConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetWallColor1String\fP ()"
.br
.ti -1c
.RI "string \fBgetWallColor2ConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetWallColor2String\fP ()"
.br
.ti -1c
.RI "string \fBgetFloorColor1ConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetFloorColor1String\fP ()"
.br
.ti -1c
.RI "string \fBgetFloorColor2ConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetFloorColor2String\fP ()"
.br
.ti -1c
.RI "string \fBgetHeadingColorConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetHeadingColorString\fP ()"
.br
.ti -1c
.RI "string \fBgetOverlayImageConfigString\fP ()"
.br
.ti -1c
.RI "string \fBgetOverlayImageString\fP ()"
.br
.ti -1c
.RI "int \fBsetParameter\fP (string parameter, string value)"
.br
.ti -1c
.RI "void \fBsetToDefaultConfiguration\fP ()"
.br
.ti -1c
.RI "bool \fBhasParameter\fP (string parameterName)"
.br
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "string \fBtheme\fP"
.br
.ti -1c
.RI "string \fBtheme_fade_color\fP"
.br
.ti -1c
.RI "string \fBtheme_wall_color_1\fP"
.br
.ti -1c
.RI "string \fBtheme_wall_color_2\fP"
.br
.ti -1c
.RI "string \fBtheme_floor_color_1\fP"
.br
.ti -1c
.RI "string \fBtheme_floor_color_2\fP"
.br
.ti -1c
.RI "string \fBtheme_heading_color\fP"
.br
.ti -1c
.RI "string \fBtheme_overlay_image\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
This class is responsible for the Theme Configuration management\&. 
.SH "Constructor & Destructor Documentation"
.PP 
.SS "\fBConfigTheme::ConfigTheme\fP ()"
.SH "Member Function Documentation"
.PP 
.SS "string \fBConfigTheme::getFadeColorConfigString\fP ()"
.SS "string \fBConfigTheme::getFadeColorString\fP ()"
.SS "string \fBConfigTheme::getFloorColor1ConfigString\fP ()"
.SS "string \fBConfigTheme::getFloorColor1String\fP ()"
.SS "string \fBConfigTheme::getFloorColor2ConfigString\fP ()"
.SS "string \fBConfigTheme::getFloorColor2String\fP ()"
.SS "string \fBConfigTheme::getHeadingColorConfigString\fP ()"
.SS "string \fBConfigTheme::getHeadingColorString\fP ()"
.SS "string \fBConfigTheme::getOverlayImageConfigString\fP ()"
.SS "string \fBConfigTheme::getOverlayImageString\fP ()"
.SS "string \fBConfigTheme::getThemeConfigString\fP ()"
.SS "string \fBConfigTheme::getThemeString\fP ()"
.SS "string \fBConfigTheme::getWallColor1ConfigString\fP ()"
.SS "string \fBConfigTheme::getWallColor1String\fP ()"
.SS "string \fBConfigTheme::getWallColor2ConfigString\fP ()"
.SS "string \fBConfigTheme::getWallColor2String\fP ()"
.SS "bool \fBConfigTheme::hasParameter\fP (stringparameterName)"
.SS "int \fBConfigTheme::setParameter\fP (stringparameter, stringvalue)"
.SS "void \fBConfigTheme::setToDefaultConfiguration\fP ()"
.SH "Member Data Documentation"
.PP 
.SS "string \fBConfigTheme::theme\fP\fC [private]\fP"
.SS "string \fBConfigTheme::theme_fade_color\fP\fC [private]\fP"
.SS "string \fBConfigTheme::theme_floor_color_1\fP\fC [private]\fP"
.SS "string \fBConfigTheme::theme_floor_color_2\fP\fC [private]\fP"
.SS "string \fBConfigTheme::theme_heading_color\fP\fC [private]\fP"
.SS "string \fBConfigTheme::theme_overlay_image\fP\fC [private]\fP"
.SS "string \fBConfigTheme::theme_wall_color_1\fP\fC [private]\fP"
.SS "string \fBConfigTheme::theme_wall_color_2\fP\fC [private]\fP"

.SH "Author"
.PP 
Generated automatically by Doxygen for FS-UAE Gui for Linux OS from the source code\&.


configtheme.cpp
/*! \file configtheme.cpp
 configtheme.cpp

 This program is free software; you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2, or (at your option)
 any later version.

 This program is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU General Public License for more details.

 You should have received a copy of the GNU General Public License
 along with this program; if not, write to the Free Software
 Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 Written by Riva Alessandro Mario, alessandromario.riva@gmail.com
 and Sanvito Davide, dsanvito90@gmail.com. */
#include "configtheme.h"
#include <QRegExp>

const string DEFAULTTHEME="";
const string DEFAULTTHEMEFADECOLOR="#3c3b37";
const string DEFAULTTHEMEWALLCOLOR="#3c3b37";
const string DEFAULTTHEMEFLOORCOLOR1="#1d1d1b";
const string DEFAULTTHEMEFLOORCOLOR2="#111110";
const string DEFAULTTHEMEHEADINGCOLOR="#f68355";
const string DEFAULTTHEMEOVERLAYIMAGE="";

ConfigTheme::ConfigTheme()
{
    setToDefaultConfiguration();
}

string ConfigTheme::getThemeConfigString()
{
    if (theme.compare(DEFAULTTHEME)==0){return "";}
    else {return "theme = " + theme;}
}

string ConfigTheme::getThemeString()
{
    return theme;
}

string ConfigTheme::getFadeColorConfigString()
{
    if (theme_fade_color.compare(DEFAULTTHEMEFADECOLOR)==0){return "";}
    else {return "theme_fade_color = " + theme_fade_color;}
}

string ConfigTheme::getFadeColorString()
{
    return theme_fade_color;
}

string ConfigTheme::getWallColor1ConfigString()
{
    if (theme_wall_color_1.compare(DEFAULTTHEMEWALLCOLOR)==0){return "";}
    else {return "theme_wall_color_1 = " + theme_wall_color_1;}
}

string ConfigTheme::getWallColor1String()
{
    return theme_wall_color_1;
}

string ConfigTheme::getWallColor2ConfigString()
{
    if (theme_wall_color_2.compare(DEFAULTTHEMEWALLCOLOR)==0){return "";}
    else {return "theme_wall_color_2 = " + theme_wall_color_2;}
}

string ConfigTheme::getWallColor2String()
{
    return theme_wall_color_2;
}

string ConfigTheme::getFloorColor1ConfigString()
{
    if (theme_floor_color_1.compare(DEFAULTTHEMEFLOORCOLOR1)==0){return "";}
    else {return "theme_floor_color_1 = " + theme_floor_color_1;}
}

string ConfigTheme::getFloorColor1String()
{
    return theme_floor_color_1;
}

string ConfigTheme::getFloorColor2ConfigString()
{
    if (theme_floor_color_2.compare(DEFAULTTHEMEFLOORCOLOR2)==0){return "";}
    else {return "theme_floor_color_2 = " + theme_floor_color_2;}
}

string ConfigTheme::getFloorColor2String()
{
    return theme_floor_color_2;
}

string ConfigTheme::getHeadingColorConfigString()
{
    if (theme_heading_color.compare(DEFAULTTHEMEHEADINGCOLOR)==0){return "";}
    else {return "theme_heading_color = " + theme_heading_color;}
}

string ConfigTheme::getHeadingColorString()
{
    return theme_heading_color;
}

string ConfigTheme::getOverlayImageConfigString()
{
    if (theme_overlay_image.compare(DEFAULTTHEMEOVERLAYIMAGE)==0){return "";}
    else {return "theme_overlay_image = " + theme_overlay_image;}
}

string ConfigTheme::getOverlayImageString()
{
    return theme_overlay_image;
}

int ConfigTheme::setParameter(string parameter, string value)
{
    QRegExp rx("#[A-Fa-f0-9]{6}");
    if(parameter.compare("theme")==0){
        theme=value;
    } else if(parameter.compare("theme_fade_color")==0){
        if (rx.exactMatch(QString::fromStdString(value))){
            theme_fade_color=value;}
        else{
            theme_fade_color=DEFAULTTHEMEFADECOLOR;
            return -1;
        }
    } else if(parameter.compare("theme_wall_color_1")==0){
        if (rx.exactMatch(QString::fromStdString(value))){
            theme_wall_color_1=value;}
        else{
            theme_wall_color_1=DEFAULTTHEMEWALLCOLOR;
            return -1;
        }
    } else if(parameter.compare("theme_wall_color_2")==0){
        if (rx.exactMatch(QString::fromStdString(value))){
            theme_wall_color_2=value;}
        else{
            theme_wall_color_2=DEFAULTTHEMEWALLCOLOR;
            return -1;
        }
    } else if(parameter.compare("theme_floor_color_1")==0){
        if (rx.exactMatch(QString::fromStdString(value))){
            theme_floor_color_1=value;}
        else{
            theme_floor_color_1=DEFAULTTHEMEFLOORCOLOR1;
            return -1;
        }
    } else if(parameter.compare("theme_floor_color_2")==0){
        if (rx.exactMatch(QString::fromStdString(value))){
            theme_floor_color_2=value;}
        else{
            theme_floor_color_2=DEFAULTTHEMEFLOORCOLOR2;
            return -1;
        }
    } else if(parameter.compare("theme_heading_color")==0){
        if (rx.exactMatch(QString::fromStdString(value))){
            theme_heading_color=value;}
        else{
            theme_heading_color=DEFAULTTHEMEHEADINGCOLOR;
            return -1;
        }
    } else if(parameter.compare("theme_overlay_image")==0){
        theme_overlay_image=value;
    }
    return 0;
}

void ConfigTheme::setToDefaultConfiguration()
{
    theme=DEFAULTTHEME;
    theme_fade_color=DEFAULTTHEMEFADECOLOR;
    theme_wall_color_1=DEFAULTTHEMEWALLCOLOR;
    theme_wall_color_2=DEFAULTTHEMEWALLCOLOR;
    theme_floor_color_1=DEFAULTTHEMEFLOORCOLOR1;
    theme_floor_color_2=DEFAULTTHEMEFLOORCOLOR2;
    theme_heading_color=DEFAULTTHEMEHEADINGCOLOR;
    theme_overlay_image=DEFAULTTHEMEOVERLAYIMAGE;
}

bool ConfigTheme::hasParameter(string parameterName)
{
    return (parameterName.compare("theme")==0)||(parameterName.compare("theme_fade_color")==0)||(parameterName.compare("theme_wall_color_1")==0)||
            (parameterName.compare("theme_wall_color_2")==0)||(parameterName.compare("theme_floor_color_1")==0)||(parameterName.compare("theme_floor_color_2")==0)||
            (parameterName.compare("theme_heading_color")==0)||(parameterName.compare("theme_overlay_image")==0);
}
